src/hardware/mixer.cpp

```diff
@@ -789,6 +794,32 @@ AudioFrame MixerChannel::ApplyCrossfeed(const AudioFrame &frame) const
 	return {a.left + b.left, a.right + b.right};
 }
 
+void MixerChannel::WakeUp()
+{
+	autosleep = {};
+
+	if (!is_enabled)
+		Enable(true);
+}
+
+void MixerChannel::AccumulateAutoSleepFrame(const AudioFrame &frame)
+{
+	autosleep.accumulator += static_cast<int16_t>(fabsf(frame.left)) +
+	                         static_cast<int16_t>(fabsf(frame.right));
+}
+
+void MixerChannel::MaybeSleep()
+{
+	constexpr uint8_t elapsed_thredhold = 250;
+	if (++autosleep.elapsed_ms < elapsed_thredhold)
+		return;
```

Comment:
Maybe I'm missing something, but I don't see how this would cork correctly. You're just calling `MaybeSleep` from a number of places, e.g. AddSamples, but callers can add more or less than 1ms worth of samples when calling that function, no? Well, at least in principle.

Are you assuming that audio is always being rendered in 1ms chunks? Or at least most of the time, so if the elapsed time measurement is a little bit off sometimes, it wouldn't be a big deal anyway? I'd agree with that, it doesn't have to be super precise if this assumption simplifies the code. But if that's the case, at least some explanatory comments are in order I think?

Reply:
Great point; yes, I'm assuming each call to add samples is about a millisecond, but sometime we add just a single sample at a time in a loop.. so that throws this out the window. 

Will try to find a place where the "millisecond's-worth" of total added samples has been satisfied, and only then add sleeper there.

Reply:
Thinking about it more, I'll probably end up adding a bit more logic to take sample rate into account.

Reply:
I think what you have there is fine; maybe just comment that assumption and mention that the timeout is just a rough figure and the actual timeout might vary. In practical terms, it doesn't matter at all if we mute the output after 250ms, or sometime even after 300 or 400ms. Up to you, really.

Reply:
Will do. Yeah..I was thinking we can theoretically shorten the sleep threshold to the shortest wavelength that's inaudible (about 50ms, if we consider sub 20 Hz inaudible?), So in the multi-hundreds is a nice wide margin away from that.

Reply:
No need to overengineer, couple of hundred of millis is fine. Even a second or two is fine, this is about turning off a channel so it doesn't consume CPU long-term. How quickly you detect the silence and turn it off is immaterial.